server/cluster/manager.go

```diff
@@ -29,13 +29,14 @@ type Manager interface {
 	// Stop must be called before manager is dropped.
 	Stop(ctx context.Context) error
 
-	CreateCluster(ctx context.Context, clusterName string, nodeCount, replicationFactor, shardTotal uint32) (*Cluster, error)
+	GetOrCreateCluster(ctx context.Context, clusterName string, nodeCount, replicationFactor, shardTotal uint32) (*Cluster, error)
```

Comment:
Add new func `GetCluster`?
If cluster was not created, we don't need to create a new one.